pandas/tests/indexes/test_base.py

```diff
@@ -237,61 +223,63 @@ def test_constructor_int_dtype_float(self, dtype):
         result = Index([0., 1., 2., 3.], dtype=dtype)
         tm.assert_index_equal(result, expected)
 
-    def test_constructor_int_dtype_nan(self):
+    @pytest.mark.parametrize("dtype,klass_or_raises", [
```

Comment:
leave these as 2 separate. generally having parameterize where one case raises is not great (you can name the same however, with _errors on one of the test names)